### PR TITLE
core: drop implicit support of PrivateUsers=off

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for fa693fdc7e17618958c505af4b2f39ecd1c3363e.

### CODE REVIEW

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
@yuwata The title is a bit misleading no, since this doesn't actually drop support for anything?

File: src/core/namespace.c [hunk 1]
```diff
@@ -3237,7 +3237,7 @@ static const char* const private_tmp_table[_PRIVATE_TMP_MAX] = {
 DEFINE_STRING_TABLE_LOOKUP_WITH_BOOLEAN(private_tmp, PrivateTmp, PRIVATE_TMP_CONNECTED);
 
 static const char* const private_users_table[_PRIVATE_USERS_MAX] = {
-        [PRIVATE_USERS_OFF]      = "off",
+        [PRIVATE_USERS_OFF]      = "no",
```
Comment (poettering) [inline src/core/namespace.c]: hmm, what's the rationale of this change? 

I am not a fan of enums whose string representation is different from their name. i.e. here now "off" vs. "no". Either rename neither or both. But I don#t grok the rationale here?

**Comment by yuwata** [comment]
@DaanDeMeyer and @poettering The commit message is updated. Also, renamed _OFF -> _NO.
PTAL.

File: src/core/namespace.c [hunk 2]
```diff
@@ -3229,7 +3229,7 @@ static const char* const private_tmp_table[_PRIVATE_TMP_MAX] = {
 DEFINE_STRING_TABLE_LOOKUP_WITH_BOOLEAN(private_tmp, PrivateTmp, PRIVATE_TMP_CONNECTED);
 
 static const char* const private_users_table[_PRIVATE_USERS_MAX] = {
-        [PRIVATE_USERS_OFF]      = "off",
+        [PRIVATE_USERS_NO]       = "no",
```
Comment (poettering) [inline src/core/namespace.c:3232]: I think "no" for this is slightly nicer than "off", but do note that parse_boolean() is considers "off" and "no" equivalent, hence this change doesn't really change much? not grokking the reasoning here?

(not opposed to the change, don't get me wrong, I think yes/no are better words than "on/off", even if we support both everywhere)
